File: 1143. Longest Common Subsequence/Solution1143.cpp
class Solution1143 {
public:
    int longestCommonSubsequence(string text1, string text2) {
        int n1 = text1.length(), n2 = text2.length();
        vector<vector<int>> dp(n1+1, vector<int>(n2+1, 0));
        for (int i = 0; i < n1; i++) {
            for (int j = 0; j < n2; j++) {
                if (text1[i] == text2[j]) {
                    dp[i+1][j+1] = max(dp[i+1][j+1], dp[i][j] + 1);
                } else {
                    dp[i+1][j+1] = max(dp[i][j+1], dp[i+1][j]);
                }
            }
        }
        return dp[n1][n2];
    }
};

File: 1143. Longest Common Subsequence/Solution1143.js
var longestCommonSubsequence = function(text1, text2) {
    let n1 = text1.length;
    let n2 = text2.length;
    let dp = new Array(n1+1).fill(null).map(()=>new Array(n2+1).fill(0));
    for (let i = 0; i < n1; i++) {
        for (let j = 0; j < n2; j++) {
            if (text1[i] == text2[j]) {
                dp[i+1][j+1] = Math.max(dp[i+1][j+1], dp[i][j] + 1);
            } else {
                dp[i+1][j+1] = Math.max(dp[i][j+1], dp[i+1][j]);
            }
        }
    }
    return dp[n1][n2];
};